test/unit/org/apache/cassandra/cql3/validation/operations/SelectTest.java

```diff
@@ -2554,6 +2554,24 @@ public void filteringOnStaticColumnTest() throws Throwable
         });
     }
 
+    @Test
+    public void filteringOnDeletedStaticColumnValue() throws Throwable {
```

Comment:
can we have a test for non-static as well?  If we delete v0 only and not v1 then the row will exist and the column would have been deleted